src/Interpreters/RewriteArrayExistsFunctionVisitor.cpp

```diff
@@ -20,21 +20,21 @@ void RewriteArrayExistsFunctionMatcher::visit(ASTPtr & ast, Data & data)
         if (join->using_expression_list)
         {
             auto * it = std::find(join->children.begin(), join->children.end(), join->using_expression_list);
+            if (it == join->children.end())
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "Could not find join->using_expression_list in '{}'", join->formatForLogging());
 
             visit(join->using_expression_list, data);
-
-            if (it && *it != join->using_expression_list)
```

Comment:
Fix 2. It shouldn't have been `it && ...` but `it != join->children.end() && ...`.

In any case, it's much better to assert your expectations (which is that the named pointer must be a child) instead of adding unexplained `if`'s